.github/workflows/agw-workflow.yml

```diff
@@ -78,3 +82,40 @@ jobs:
           fetch-depth: 0
       - name: jsonlint-mconfig
         run: find . -name gateway.mconfig -print0 | xargs --max-args=1 --null --replace='%' sh -c ">/dev/null jq . % || { echo % is not a valid json file; exit 1; } "
+
+  pylint:
+    name: pylint
+    runs-on: ubuntu-20.04
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_not_skip == 'true' }}
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
```

Comment:
As discussed: The `ref:` and `repository:` attributes are not needed here and on push trigger might be empty.